core/sail/inferencer/src/main/java/org/eclipse/rdf4j/sail/inferencer/fc/FastRdfsForwardChainingSail.java

```diff
@@ -0,0 +1,209 @@
+package org.eclipse.rdf4j.sail.inferencer.fc;
+
+import org.eclipse.rdf4j.common.iteration.Iterations;
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+import org.eclipse.rdf4j.repository.Repository;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+import org.eclipse.rdf4j.repository.RepositoryResult;
+import org.eclipse.rdf4j.sail.NotifyingSail;
+import org.eclipse.rdf4j.sail.SailException;
+import org.eclipse.rdf4j.sail.inferencer.InferencerConnection;
+import org.eclipse.rdf4j.sail.inferencer.fc.AbstractForwardChainingInferencer;
+
+import java.io.File;
+import java.util.*;
+
+/**
+ * @author Håvard Mikkelsen Ottestad
+ */
+
+public class FastRdfsForwardChainingSail extends AbstractForwardChainingInferencer {
+
+    NotifyingSail data;
+    Repository schema;
+
+    boolean useAllRdfsRules = true;
+
+    List<Statement> subClassOfStatemenets = new ArrayList<>();
+    List<Statement> propertyStatements = new ArrayList<>();
+    List<Statement> subPropertyOfStatemenets = new ArrayList<>();
+    List<Statement> rangeStatemenets = new ArrayList<>();
+    List<Statement> domainStatemenets = new ArrayList<>();
```

Comment:
propertyStatements does keep everything in memory, I'll get that sorted. 

Other than that the only thing kept in memory is the Tbox.